dockerfile_parse/parser.py

```diff
@@ -549,3 +532,28 @@ def _add_instruction(self, instruction, value):
             lines = self.lines
             lines += new_line
             self.lines = lines
+
+    @property
+    def context_structure(self):
+        """
+        :return: list of context objects
+            (Contains info about labels and environment variables for each line.)
+        """
+        instructions = []
+        envs = {}
+        labels = {}
```

Comment:
Thank you for your response. Is this better?

```python
@property
    def context_structure(self):
        """
        :return: list of context objects
            (Contains info about labels and environment variables for each line.)
        """
        instructions = []
        last_context = Context()
        for instr in self.structure:
            context = Context(envs=dict(last_context.envs),
                              labels=dict(last_context.labels))
            instruction_type = instr['instruction']
            if instruction_type in ["ENV", "LABEL"]:
                val = get_key_val_dictionary(instruction_value=instr['value'],
                                             env_replace=self.env_replace,
                                             envs=last_context.envs)
                context.update_line_value(context_type=instruction_type, value=val)
                context.envs.update(context.line_envs)
                context.labels.update(context.line_labels)

            instructions.append(context)
            last_context = context
        return instructions
```

And what about updating of `context.envs`/`context.labels` in `context.update_line_value`? So we can remove these lines:
```python
context.envs.update(context.line_envs)
context.labels.update(context.line_labels)
```